core/client/src/main/java/alluxio/client/file/BaseFileSystem.java

```diff
@@ -86,6 +91,12 @@ public void createDirectory(AlluxioURI path, CreateDirectoryOptions options)
     try {
       masterClient.createDirectory(path, options);
       LOG.debug("Created directory {}, options: {}", path.getPath(), options);
+    } catch (AlreadyExistsException e) {
```

Comment:
Compatibility - the user API says that it throws `AlluxioException` when an error occurs in Alluxio